ably/ably-chat-js
# react-hooks: Add useOccupancy hook

### Context

* [CHA-389]
* As part of the react hooks development, a new hook to interact with the `Occupancy` instance of a room was required.

### Description

* Added the `useOccupancy` react hook to interact with the `Occupancy` instance of a room. This allows the user to interact with the underlying `Occupancy` instance, receive connection and presence metrics, subscribe to occupancy events, and throttle the update rate of the occupancy metrics.
  * Updated the readme to document hook usage.
  * Added unit tests for the new hook.
  * Added integration tests for the new hook.
* Added lodash throttle function.
* Updated the demo app to use the new hook.
* Added `StatusParams` and `ChatStatusResponse` to the README.md to avoid repetition in the documentation.


### Checklist


### Testing Instructions (Optional)

1. Spin up a local instance of the demo app and navigate to the `OccupancyComponent`. 
2. Verify that the `Occupancy` metrics are displayed correctly and that the throttle functionality is working as expected. 
3. Try changing the throttle length (note: there is a minimum update frequency from `Occupancy`)



[CHA-389]: https://ably.atlassian.net/browse/CHA-389?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
